Subject: [PATCH] gitweb: use an absolute base URL

That is one starting with a '/', NOT one starting
with host and port.  Yikes!

There's no reason to embed the scheme, host and
port number when an absolute URL will do just fine.

Signed-off-by: Kyle J. McKay <mackyle@gmail.com>
---
gitweb/gitweb.perl | 5 +++--
1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/gitweb/gitweb.perl b/gitweb/gitweb.perl
index 52bf96b0..11474463 100755
--- a/gitweb/gitweb.perl
+++ b/gitweb/gitweb.perl
@@ -45,7 +45,8 @@ sub evaluate_uri {
 
 	# Base URL for relative URLs in gitweb ($logo, $favicon, ...),
 	# needed and used only for URLs with nonempty PATH_INFO
-	our $base_url = $my_url;
+	# This must be an absolute URL (i.e. no scheme, host or port), NOT a full one
+	our $base_url = $my_uri || '/';
 
 	# When the script is used as DirectoryIndex, the URL does not contain the name
 	# of the script file itself, and $cgi->url() fails to strip PATH_INFO, so we
@@ -69,7 +70,7 @@ sub evaluate_uri {
 		if ($my_url =~ s,\Q$path_info\E$,, &&
 		    $my_uri =~ s,\Q$path_info\E$,, &&
 		    defined $ENV{'SCRIPT_NAME'}) {
-			$base_url = $cgi->url(-base => 1) . $ENV{'SCRIPT_NAME'};
+			$base_url = $ENV{'SCRIPT_NAME'} || '/';
 		}
 	}
 
---
